src/vsphere_cpi/spec/unit/cloud/vsphere/attach_tag_to_vm_spec.rb

```diff
@@ -0,0 +1,402 @@
+require 'spec_helper'
+require 'vsphere-automation-cis'
+require 'vsphere-automation-vcenter'
+
+module VSphereCloud
+  describe TaggingTag::AttachTagToVm, fake_logger: true do
+    subject(:api_client) { VSphereAutomation::ApiClient.new }
+    subject(:tagging_tag) { TaggingTag::AttachTagToVm.new(api_client) }
+
+    describe '#retrieve_category_id' do
+      let(:result_1) do
+        instance_double(VSphereAutomation::CIS::CisTaggingCategoryResult,
+          value: instance_double(VSphereAutomation::CIS::CisTaggingCategoryModel,
+           id: "fake-category-id-1",
+           name: "fake-category-name-1"
+          )
+        )
+      end
+      let(:result_2) do
+        instance_double(VSphereAutomation::CIS::CisTaggingCategoryResult,
+          value: instance_double( VSphereAutomation::CIS::CisTaggingCategoryModel,
+            id: "fake-category-id-2",
+            name: "fake-category-name-2"
+          )
+        )
+      end
+      let(:category_ids) { %w[ fake-category-id-1 fake-category-id-2 ] }
+
+      context 'when try to find a category exists on vCenter' do
```

Comment:
`when trying to find a category that exists on vCenter`

Reply:
Done.